File: src/main/java/com/dh/ondot/schedule/api/request/QuickScheduleCreateRequest.java
package com.dh.ondot.schedule.api.request;

import jakarta.validation.Valid;
import jakarta.validation.constraints.*;

import java.time.LocalDateTime;

public record QuickScheduleCreateRequest(
        @NotNull LocalDateTime appointmentAt,

        @NotNull @Valid PlaceDto departurePlace,

        @NotNull @Valid PlaceDto arrivalPlace
) {
}

File: src/main/kotlin/com/dh/ondot/schedule/presentation/request/QuickScheduleCreateRequest.kt
package com.dh.ondot.schedule.presentation.request

import jakarta.validation.Valid
import jakarta.validation.constraints.NotNull
import java.time.LocalDateTime

data class QuickScheduleCreateRequest(
    @field:NotNull
    val appointmentAt: LocalDateTime,

    @field:NotNull @field:Valid
    val departurePlace: PlaceDto,

    @field:NotNull @field:Valid
    val arrivalPlace: PlaceDto,
)
